File: app/schemas/report.py
from datetime import datetime

from pydantic import BaseModel


class ReportTimeRange(BaseModel):
    date_from: str
    date_to: str
    granularity: str = "day"


class ReportMetric(BaseModel):
    key: str
    label: str
    value: int | float
    unit: str = ""


class ReportSeriesPoint(BaseModel):
    date: str
    order_count: int = 0
    gmv_cent: int = 0
    refund_amount_cent: int = 0


class ReportNameValue(BaseModel):
    name: str
    value: int | float
    id: int | None = None
    amount_cent: int | None = None


class ReportTopProduct(BaseModel):
    product_id: int
    product_name: str
    quantity: int
    amount_cent: int


class ReportOverviewResponse(BaseModel):
    scope: str
    scope_id: int | None = None
    time_range: ReportTimeRange
    generated_at: datetime
    summary: list[ReportMetric]
    sales_trend: list[ReportSeriesPoint]
    order_status: list[ReportNameValue]
    top_products: list[ReportTopProduct]
    top_merchants: list[ReportNameValue]
    refund_status: list[ReportNameValue]
    promotion_summary: list[ReportMetric]
    community_summary: list[ReportMetric]

File: admin-frontend/src/services/report.ts
import { http } from './http'

export type ReportMetric = {
  key: string
  label: string
  value: number
  unit: string
}

export type ReportSeriesPoint = {
  date: string
  order_count: number
  gmv_cent: number
  refund_amount_cent: number
}

export type ReportNameValue = {
  id?: number | null
  name: string
  value: number
  amount_cent?: number | null
}

export type ReportTopProduct = {
  product_id: number
  product_name: string
  quantity: number
  amount_cent: number
}

export type ReportOverview = {
  scope: 'platform' | 'merchant'
  scope_id?: number | null
  time_range: {
    date_from: string
    date_to: string
    granularity: string
  }
  generated_at: string
  summary: ReportMetric[]
  sales_trend: ReportSeriesPoint[]
  order_status: ReportNameValue[]
  top_products: ReportTopProduct[]
  top_merchants: ReportNameValue[]
  refund_status: ReportNameValue[]
  promotion_summary: ReportMetric[]
  community_summary: ReportMetric[]
}

export function getPlatformReportOverview() {
  return http.get('/admin/reports/platform/overview', { headers: { 'X-Admin-Session': 'platform' } })
}

export function getMerchantReportOverview() {
  return http.get('/admin/reports/merchant/overview', { headers: { 'X-Admin-Session': 'merchant' } })
}
